packages/driver/src/cypress/cy.ts

```diff
@@ -860,8 +846,15 @@ export class $Cy extends EventEmitter2 implements ITimeouts, IStability, IAssert
         return memo
       }, [initialCommand])
 
+      const chainerId = this.state('chainerId')
```

Comment:
Here is the change which fixes the todomvc regression. The source of the error was very, very distant from the error that was ultimately thrown.

In the old code, `replayCommandFrom()` *updates the chainerId before cloning* - it updates the original command. This was causing a chain of unintended side-effects, ultimately resulting in an `undefined` subject, rather than the one properly stored in `cy.state('subjects')`.

. In the new version, it clones and *then* updates, leaving the original command intact. Cypress no longer gets confused when replaying commands.

----

I'm not terribly fussed about the exact details here - this entire "replayCommandsFrom" logic is going to be removed completely as part of the Detached DOM work. As long as it keeps working in the interim (and we have a *lot* of tests ensuring it does), I'm not much fussed about exactly what it looks like.